test/simulator/misc/random-teams.js

```diff
@@ -128,13 +138,27 @@ describe(`Factory sets`, function () {
 							assert(ability.name === abilityName, `miscapitalized ability "${abilityName}" of ${species}`);
 						}
 
+						for (const natureName of [].concat(set.nature)) {
+							const nature = Dex.getNature(natureName);
+							assert(nature.exists, `invalid nature "${natureName}" of ${species}`);
+							assert(nature.name === natureName, `miscapitalized nature "${natureName}" of ${species}`);
+						}
+
 						for (const moveSpec of set.moves) {
 							for (const moveName of [].concat(moveSpec)) {
```

Comment:
same odd `[].concat` pattern - what am I missing?

Reply:
I wish I had an answer for this, but I didn't write it, sorry